libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -550,6 +573,74 @@ bool AC_Fence::pre_arm_check(char *failure_msg, const uint8_t failure_msg_len) c
     return true;
 }
 
+/*
+  get our altitude in the FENCE_ALT_TYPE frame
+  return false if not available
+ */
+bool AC_Fence::get_alt_U_in_frame(float &alt, Location::AltFrame alt_frame) const
+{
+    const auto &ahrs = AP::ahrs();
```

Comment:
Why aren't you just getting the current location and then asking that for your height in the frame you want?